examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchor.agent.ts

```diff
@@ -0,0 +1,69 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { INVALID_ANCHOR_HASH_INPUT } from '../api/err.messages';
+import { v4 as uuidv4 } from 'uuid';
+import { CCSMAnchor } from '../models/ccsmAnchor';
+import { CCSMAnchorStorageAgent } from './ccsmAnchorStorage.agent';
+import { BpiAccount } from '../../identity/bpiAccounts/models/bpiAccount';
+import { DocumentObject } from '../models/document';
+import { ZeroKnowledgeProofVerificationInput } from '../models/zeroKnowledgeProofVerificationInput';
+import { BpiSubjectAccount } from '../../identity/bpiSubjectAccounts/models/bpiSubjectAccount';
+@Injectable()
+export class CCSMAnchorAgent {
+  constructor(private readonly storageAgent: CCSMAnchorStorageAgent) {}
+
+  public throwErrorIfCCSMAnchorInputInvalid(
+    inputForProofVerification:
+      | DocumentObject
+      | ZeroKnowledgeProofVerificationInput,
+  ): void {
+    if (
+      inputForProofVerification instanceof DocumentObject &&
+      (!inputForProofVerification.documentObjectInput ||
+        !inputForProofVerification.documentObjectType)
+    ) {
+      throw new BadRequestException(INVALID_ANCHOR_HASH_INPUT);
+    }
+  }
+
+  public createNewCCSMAnchor(
+    owner: BpiSubjectAccount,
+    agreementState: BpiAccount,
+    document: DocumentObject,
+    signature: string,
+  ): CCSMAnchor {
+    const hash =
+      this.convertDocumentToHashAndThrowErrorIfDocumentValidationFails(
```

Comment:
convertDocumentToHashAndThrowErrorIfDocumentValidationFails should be called directly from the command
createNewCCSMAnchor should only generate the new object - single responsibility is best whenever possible